File: November 2022/1047. Remove All Adjacent Duplicates In String/solution.java
// Asked on 10-11-2022 (DD-MM-YYYY)
// Language: Java

class Solution {
  public String removeDuplicates(String s) {
    Stack < Character > stack = new Stack < > ();
    StringBuilder sb = new StringBuilder();
    char[] arr = s.toCharArray();
    for (char i: arr) {
      if (!stack.isEmpty() && stack.peek() == i) {
        stack.pop();
      } else {
        stack.push(i);
      }
    }

    for (char i: stack) {
      sb.append(i);
    }

    return sb.toString();
  }
}

File: November 2022/1047. Remove All Adjacent Duplicates In String/solution.py
# Asked on 10-12-2022 (DD-MM-YYYY)
# Language: Python

class Solution:
    def removeDuplicates(self, s: str) -> str:
        stack = []
        
        for i in s:
            if(len(stack)!=0 and stack[-1]==i):
                stack.pop()
            else:
                stack.append(i)
        return ''.join(stack)
      
      
